tests/test_osut.py

```diff
@@ -5187,6 +5187,102 @@ def test34_generated_skylight_wells(self):
+
+        # TOTAL plenum roof area (4x surfaces), no overhangs.
+        roofs = osut.facets(plenum, "Outdoors", "RoofCeiling")
+        total = sum([ruf.grossArea() for ruf in roofs])
+        self.assertAlmostEqual(total, 82.21, places=2)
+
+        # A single plenum above all 4 occupied rooms. Reports same GRA.
+        gra_seb1 = osut.grossRoofArea(model.getSpaces())
+        gra_seb2 = osut.grossRoofArea(entry)
+        self.assertAlmostEqual(gra_seb1, gra_seb2, places=2)
+        self.assertAlmostEqual(gra_seb1, total, places=2)
+
+        sky_area = srr * total
+
+        # Before adding skylight wells.
+        if version >= 350:
+            for sp in [plenum, entry, office, open, utility]:
+                self.assertTrue(sp.isEnclosedVolume())
+                self.assertTrue(sp.isVolumeDefaulted())
+                self.assertTrue(sp.isVolumeAutocalculated())
+                self.assertGreater(sp.volume(), 0)
+
+                zn = sp.thermalZone()
+                self.assertTrue(zn)
+                zn = zn.get()
+                self.assertTrue(zn.isVolumeDefaulted())
+                self.assertTrue(zn.isVolumeAutocalculated())
+                self.assertFalse(zn.volume())
+
+        # The method returns the GRA, calculated BEFORE adding skylights/wells.
+        rm2 = osut.addSkyLights(model.getSpaces(), dict(area=sky_area))
+        if o.logs(): print(o.logs())
+        self.assertAlmostEqual(rm2, total, places=2)
+
+        entry_skies   = osut.facets(entry, "Outdoors", "Skylight")
+        office_skies  = osut.facets(office, "Outdoors", "Skylight")
+        utility_skies = osut.facets(utility, "Outdoors", "Skylight")
+        open_skies    = osut.facets(open, "Outdoors", "Skylight")
+
+        self.assertFalse(entry_skies)
+        self.assertFalse(office_skies)
+        self.assertFalse(utility_skies)
+        self.assertEqual(len(open_skies), 1)
+        open_sky = open_skies[0]
+
+        skm2 = open_sky.grossArea()
+        self.assertAlmostEqual(skm2 / rm2, srr, places=2)
+
+        # Assign construction to new skylights.
+        construction = osut.genConstruction(model, dict(type="skylight", uo=2.8))
+        self.assertTrue(open_sky.setConstruction(construction))
+
+        # No change after adding skylight wells.
+        if version >= 350:
+            for sp in [plenum, entry, office, open, utility]:
+                self.assertTrue(sp.isEnclosedVolume())
+                self.assertTrue(sp.isVolumeDefaulted())
+                self.assertTrue(sp.isVolumeAutocalculated())
+                self.assertGreater(sp.volume(), 0)
+
+                zn = sp.thermalZone()
+                self.assertTrue(zn)
+                zn = zn.get()
+                self.assertTrue(zn.isVolumeDefaulted())
+                self.assertTrue(zn.isVolumeAutocalculated())
+                self.assertFalse(zn.volume())
+
+        model.save("./tests/files/osms/out/seb_sky.osm", True)
```

Comment:
The _addSkylights_' built-in criteria filters will favour adding a single skylight _strip_ (+ well) in the largest room:

<img width="858" height="680" alt="seb_sky" src="https://github.com/user-attachments/assets/9f85d332-44ac-4f5e-8a18-1af84416ddcb" />

Moving along ...